metals/src/main/scala/scala/meta/internal/metals/MetalsEnrichments.scala

```diff
@@ -352,6 +380,40 @@ object MetalsEnrichments
         }
         AbsolutePath(out)
       }
+
+      if (path.toNIO.getFileSystem == workspace.toNIO.getFileSystem) {
+        path
+      } else {
+        path.jarPath match {
+          case Some(jar) =>
+            val jarDir =
+              workspace.resolve(Directories.dependencies).resolve(jar.filename)
+            val out = jarDir.resolveZipPath(path.toNIO)
+            val jarMetaFile = jarDir.resolve("jar.meta")
+
+            lazy val currentJarMeta = readJarMeta(jarMetaFile)
+            lazy val jarMeta = toJarMeta(jar)
+
+            val updateMeta = !jarDir.exists || !currentJarMeta.contains(jarMeta)
+            if (!out.exists || updateMeta) {
```

Comment:
Wouldn't it be enough to just check if the file exists and then not copy it? Since, we have directory based on the jar name and jar names contain the version, it should never really change, only if someone creates snapshot jars locally, but in that case they will also look at the original sources.

Reply:
I agree that it feels a bit of redundantly but this mechanic with jar-name is far from being perfect. 

There might be a lot of rare cases when jar name won't guarantee that it's the same artifact.
In addition to local snapshots it might be a snapshot jar from repository. Or it might be a forked artifact with the same version (like: `org.scalameta.metals-..` and `com.virtuslab.metals-..`).

I think that even in these rare cases, it is worth preventing the use of the wrong sources, if possible. 
Otherwise someone might be confused/spend time investigating why the sources is different from a real execution before he realize that it's outdated and only manual cleanup will help.

Reply:
Makes sense!